nodeJS/testing_express/testing_database_operations.md

```diff
@@ -1,86 +1,70 @@
 
-Setting it up is straightforward, but there are a few things you need to do.  First, in your actual app, you need to move your mongo/mongoose setup into its own file as seen in the example below.
+We'll leverage environment variables for your codebase to identify which database to use. First, add the test database in your `.env` file:
 
-```javascript
-//// mongoConfig.js
-const mongoose = require("mongoose");
-
-const mongoDb = `YOUR MONGO URL`;
-
-mongoose.connect(mongoDb);
-const db = mongoose.connection;
-db.on("error", console.error.bind(console, "mongo connection error"));
+```text
+NODE_ENV=development
+DATABASE_URL=postgresql://<user>:<password>@localhost:3306/inventory_application
+TEST_DATABASE_URL=postgresql://<user>:<password>@localhost:3306/test_inventory_application
 ```
 
-The above code should look very familiar to you by now... it's the same setup code we've been using all along. The only difference is that we've moved it out to its own file so that in our test files we can use a *different* config file that sets up mongodb-memory-server for us. All you have to do now is `require("./mongoConfig")` in your `app.js` file.
+Next, setup appropriate npm scripts in your `package.json` file:
 
-Next we need to create a separate config for our testing environment. The config file that you can find on the `mongodb-memory-server` repo README should work just fine. Below is a slightly edited version of it. Copy this to a new file called `mongoConfigTesting.js`
-
-```javascript
-//// mongoConfigTesting.js 
-const mongoose = require("mongoose");
-const { MongoMemoryServer } = require("mongodb-memory-server");
-
-async function initializeMongoServer() {
-  const mongoServer = await MongoMemoryServer.create();
-  const mongoUri = mongoServer.getUri();
-
-  mongoose.connect(mongoUri);
-
-  mongoose.connection.on("error", e => {
-    if (e.message.code === "ETIMEDOUT") {
-      console.log(e);
-      mongoose.connect(mongoUri);
-    }
-    console.log(e);
-  });
-
-  mongoose.connection.once("open", () => {
-    console.log(`MongoDB successfully connected to ${mongoUri}`);
-  });
+```json
+{
+  // other stuff
+  "scripts": {
+    "dev": "NODE_ENV=development && node app.js",
+    "test": "NODE_ENV=test && jest"
+  },
+  // even more stuff
 }
-
-module.exports = initializeMongoServer;
 ```
 
-Now, if your tests are set up similarly to the tests in our last project, you can call this function in your testing file, and then any operations that work on your mongo database will use this testing one instead of your real one.
+Based on the `NODE_ENV`, you can programmatically switch out database urls:
+
+```javascript
+const databaseUrl = process.env.NODE_ENV === 'test' ? process.env.TEST_DATABASE_URL : process.env.DATABASE_URL;
```

Comment:
Can format this way to prevent horizontal scrolling.

```suggestion
const databaseUrl = process.env.NODE_ENV === 'test'
  ? process.env.TEST_DATABASE_URL
  : process.env.DATABASE_URL;
```